packages/core/utils/src/dml/entity-builder.ts

```diff
@@ -162,7 +162,13 @@ export class EntityBuilder {
    *   relationship requires a pivot table to establish a many to many
    *   relationship between two entities
    */
-  manyToMany<T>(entityBuilder: T, options?: RelationshipOptions) {
+  manyToMany<T>(
+    entityBuilder: T,
+    options?: RelationshipOptions & {
+      pivotTable?: string
+      pivotEntity?: () => DmlEntity<any>
+    }
```

Comment:
```suggestion
    options?: RelationshipOptions & ({
      pivotTable?: string
      pivotEntity?: never
    } | {
      pivotTable?: never
      pivotEntity?: () => DmlEntity<any>
    })
```

The idea being that we only allow one or the other wdyt? Unless it makes sense to have both but not sure

Reply:
What if I want to use a custom pivot table name with a custom pivot entity. Remember, we do not allow defining a custom table name via entities. So there is no way for me to configure that on the pivot entity

Reply:
yes thats true, but don't we want to have the same behaviour has any other entity? meaning that the table name will be constructed from the entity name, here the pivot entity name would be the one defining this table name. If you look at the migration level it is also probably what will happen, a table will be created for the pivot entity which might in that case defer from configuration

Reply:
Validated through a call discussion 💪

Reply:
Makes sense!